api/graphql/reservations/recurring_reservation_serializers/create_serializers.py

```diff
@@ -0,0 +1,124 @@
+        source="ability_group",
+        allow_null=True,
+        required=False,
+    )
+
+    weekdays = ValidatingListField(
+        child=ChoiceIntegerField(choices=WEEKDAYS.CHOICES, allow_null=False),
+        allow_empty=False,
+        required=True,
+        help_text="List of weekdays which days the reservations occurs",
+    )
+
+    name = serializers.CharField(required=False, default="")
+    begin_time = serializers.TimeField(
+        required=True, help_text="Time when reservations begins."
+    )
+    end_time = serializers.TimeField(
+        required=True, help_text="Time when reservations ends."
+    )
+    begin_date = serializers.DateField(
+        required=True, help_text="Date when first reservation begins."
+    )
+    end_date = serializers.DateField(
+        required=True, help_text="Date when last reservation begins."
+    )
+    interval = serializers.IntegerField(required=True)
+
+    class Meta:
+        model = RecurringReservation
+        fields = [
+            "pk",
+            "user",
+            "name",
+            "reservation_unit_pk",
+            "age_group_pk",
+            "ability_group_pk",
+            "interval",
+            "weekdays",
+            "begin_time",
+            "end_time",
+            "begin_date",
+            "end_date",
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["user"].readonly = True
+
+    def validate(self, data):
+        begin = data.get("begin_time", getattr(self.instance, "begin_time", None))
+        end = data.get("end_time", getattr(self.instance, "end_time", None))
+
+        if end < begin:
+            raise ValidationErrorWithCode(
+                "Begin time cannot be after end time.",
+                ValidationErrorCodes.RESERVATION_BEGIN_AFTER_END,
+            )
+
+        begin_date = data.get("begin_date", getattr(self.instance, "begin_date", None))
+        end_date = data.get("end_date", getattr(self.instance, "end_date", None))
+
+        if end_date < begin_date:
+            raise ValidationErrorWithCode(
+                "Begin date cannot be after end date.",
+                ValidationErrorCodes.RESERVATION_BEGIN_AFTER_END,
+            )
+
+        user = self.context.get("request").user
+        if user.is_anonymous:
+            user = None
+
+        data["user"] = user
+
+        return data
+
+    def validate_weekdays(self, weekdays):
+        days = []
+        weekdays = weekdays or getattr(self.instance, "weekdays", [])
+        for weekday in weekdays:
+            if weekday not in WEEKDAYS.VALUES:
+                raise ValidationErrorWithCode(
+                    f"Invalid weekday {weekday}.", ValidationErrorCodes.INVALID_WEEKDAY
+                )
+            days.append(str(weekday))
+
+        return ",".join(days)
+
+    def validate_interval(self, interval):
+        allowed_values = [i for i in range(170) if i % 7 == 0]
```

Comment:
`170` feels like a magic number. Where does it come from?

Reply:
Nice catch, it is a magic number 🪄 indeed! I'll need to change this a bit and ask is there any max value for it.

Reply:
Removed the magic number and also renamed the field to `recurring_in_days`